src/client/components/SupplierSearch/SupplierSearch.react.js

```diff
@@ -0,0 +1,111 @@
+import React, { Component, PropTypes } from 'react';
+import SupplierEditorLocales from './../SupplierEditor/i18n';
+import ReactTable from 'react-table';
+import 'react-table/react-table.css';
+import request from 'superagent-bluebird-promise';
+import SupplierConstraints from '../../utils/validatejs/supplierConstraints';
+import DisplayCountryTableField from '../DisplayTable/DisplayCountryTableField.react';
+require('./SupplierSearch.css');
+
+export default class SupplierSearch extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      form: {
+        keyword: '',
+      },
+      data: []
+    }
+  }
+
+  static propTypes = {
+    actionUrl: React.PropTypes.string.isRequired
+  };
+
+  static contextTypes = {
+    i18n : React.PropTypes.object.isRequired,
+    showNotification: React.PropTypes.func
+  };
+
+  componentWillMount(){
+    this.constraints = new SupplierConstraints(this.context.i18n);
+    this.context.i18n.register('SupplierEditorLocales', SupplierEditorLocales);
+  }
+
+  searchSupplier() {
+    const getRequest = request.get(`${this.props.actionUrl}/supplier/api/suppliers?search=${this.refs.keyword.value}`);
+    let result = getRequest.set('Accept', 'application/json').promise();
+    result.then((data) => {
+      const oldState = this.state;
+      const newState = Object.assign({}, oldState, {
+        form: {
+          keyword: this.refs.keyword.value
+        },
+        data: data.body
+      });
+      this.setState(newState);
+    });
+  }
+
+  renderSearchBox() {
+    return (<div className="form-group search-box">
+      <label className="col-xs-12 col-sm-6 col-md-4 control-label">Search by</label>
```

Comment:
Translate label. It should be:
- `en: 'Search by name, city, VAT id, Tax id, registration number, DUNS number, or Global Location Number'`
- `de: 'Suche nach Name, Stadt, Umsatzsteuer-ID, Steuernummer, Registrierungsnummer, DUNS-Nummer oder Global Location Number'`